HTX stress test for NIC devices.

pridhiviraj opened the pull request:
This test as of now assumes htx is installed in both Host & Peer.
And this test will be of two phases

Phase 1: Configure all necessary pre-setup steps for both the n/w
         interfaces in both Host & Peer
Phase 2: Start the HTX setup & execution of test for a time_limit
         and monitor's the HTX error log for any errors in both
         Host & Peer

Signed-off-by: Pridhiviraj Paidipeddi <ppaidipe@linux.vnet.ibm.com>

narasimhan-v commented on io/net/htx_nic_devices.py.data/htx_nic_devices.yaml:
```diff
@@ -0,0 +1,11 @@
+peer_ip: y.yy.yyy.yy
+peer_user: "root"
+peer_password: "******"
+host_intf1: "enp1s0f0"
```
Please use the format of parameters as other tests.
In this case, it is `host_interfaces` and `peer_interfaces`

narasimhan-v commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        self.peer_password = self.params.get("peer_password",
+                                             '*', default=None)
+        self.host_intf1 = self.params.get("host_intf1", '*', default=None)
+        self.host_intf2 = self.params.get("host_intf2", '*', default=None)
```
Why are we assuming we will have 2 interfaces ? We can generalise this for 'n' interfaces, by getting a list of interfaces in the yaml file, as mentioned in the other comment.

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        '''
+        SSH Login method for remote server
+        '''
+        p = pxssh.pxssh()
```
can we have valid variable name at least length of three ?

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on lpar %s", command)
+        c = self.pxssh
```
please here as well

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+                      %s file", hosts_file)
+        with open(hosts_file, 'r') as file:
+            filedata = file.read()
+        file.close()
```
The `with` already closes the `file`, please remove the unnecessary double-close.

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        with open(hosts_file, 'r') as file:
+            filedata = file.read()
+        file.close()
+        search_str1 = host_name + ".* " + self.host_ip
```
concatenating string using `+` not looks good please use this way 

`search_str1 ='%s.* %s" %(host_name,self.host_ip ) `

PraveenPenguin replied:
and other places as well where it is applicable

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        with open(hosts_file, 'w') as file:
+            for line in filedata:
+                file.write(line)
+        file.close()
```
the same here.  double-close of file object

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        # Update other id's in host lpar
+        with open(self.bpt_file, 'r') as file:
+            filedata = file.read()
+        file.close()
```
here as well

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        with open(self.bpt_file, 'w') as file:
+            for line in filedata:
+                file.write(line)
+        file.close()
```
here as well

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        # Update net id in host lpar
+        with open(self.bpt_file, 'r') as file:
+            filedata = file.read()
+        file.close()
```
here as well

PraveenPenguin commented on io/net/htx_nic_devices.py:
```diff
@@ -0,0 +1,591 @@
+        with open(self.bpt_file, 'w') as file:
+            for line in filedata:
+                file.write(line)
+        file.close()
```
here as well

pridhiviraj commented:
Thanks @PraveenPenguin & @narasimhan-v for the review. Will work on it and send a V2.

PraveenPenguin commented:
@pridhiviraj can you please provide job log

pridhiviraj commented:
@PraveenPenguin We are doing few more fixes based on execution from different machine/distro combinations. Once it is done, a V3 with job log will attach here. Thanks

PraveenPenguin commented:
@pridhiviraj sure make sense  thanks

pridhiviraj commented:
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1114252/job.txt)

pridhiviraj commented:
Changes in V4:
            - If build_net fails, based upon the interface conenctions
              it may take time to pass pingum test. so given a 10sec's
              sleep to run the pingum test.
            - Added the README file
    
Changes in V3:
            - Fixed pingum test order, execute build_net first, if that
              fails, run pingum test. if this also fails, fail the test.
    
Changes in V2:
            - Instead of two interfaces, handled for n interfaces
            - Changed string concatenation using % operator
            - Removed duplicate file close methods
            - Fixed some other string names

hathyaga commented:
[HTX_LATEST_RUN_PASS.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1120138/HTX_LATEST_RUN_PASS.txt)

pridhiviraj commented:
Changes in V5:
	    - Updated the README with net_id limits to be used(net_id >=100 or <= 223)

narasimhan-v commented on io/net/htx_nic_devices.py.data/htx_nic_devices.yaml:
```diff
@@ -0,0 +1,9 @@
+mdt_file: 'net.mdt'
+host_ip: x.xx.xxx.xx
```
What is the need for this parameter, when we have `host_interfaces` ?

pridhiviraj replied:
this is public ip, which we need to update in both host & peer machines on the file (/etc/hosts & bpt files)
host_interfaces are can be any interfaces where we execute the htx stress, and these interfaces are different than the main interface(Which will have the public IP).

narasimhan-v replied:
For my understanding, is updating the test IPs alone not enough ?

pridhiviraj replied:
We need one peer ip(test interface ip) for executing ssh commands on peer, and also public IP's of both Host & Peer we need to update those in /etc/hosts & bpt files.

narasimhan-v commented on io/net/htx_nic_devices.py.data/htx_nic_devices.yaml:
```diff
@@ -0,0 +1,9 @@
+host_ip: x.xx.xxx.xx
+peer_ip: x.xx.xxx.xy
+peer_user: "root"
+peer_password: "******"
```
In all other tests, we have mandated passwordless ssh to be set up. Can that be followed here also please

pridhiviraj replied:
I need to check how feasible to is. Otherwise we will end up in doing lot of code changes.  I know password less ssh means there is no password, Just simply asking Is there any default password if we setup the password less ssh?

narasimhan-v replied:
Agreed, but since we run tests one after the other, and other are in a specific format, it would be good to follow the same format :)

PraveenPenguin commented:
@narasimhan-v thanks for review
